Management Group Diagnostic Settings Enablement - New Module (#321)
Azure/ALZ-Bicep: infra-as-code/bicep/modules/mgDiagSettings/README.md

# Module: Enable Diagnostic Settings on Management Groups

> This module enables Diagnostic Settings on a management group to an existing Azure Log Analytics Workspace, normally deployed during the logging.bicep module.

Reviewer: ```suggestion
This module enables the supported Diagnostic Settings categories on a Management Group to an existing Azure Log Analytics Workspace.
```

> 

Reviewer: ```suggestion
> Consider using the `mgDiagSettingsAll` orchestration module instead to simplify configuring the Diagnostic Settings for all your Management Group hierarchy in a single module. [infra-as-code/bicep/orchestration/mgDiagSettingsAll](https://github.com/Azure/ALZ-Bicep/tree/main/infra-as-code/bicep/orchestration/mgDiagSettingsAll)

```

## Parameters

> The module requires the following input parameters. This parameter is passed from the mgDiagSettingsAll.bice module.

Reviewer: ```suggestion
The module requires the following input parameters.
```

| Parameter                             | Type   | Description                                                                                                                                                                          | Requirements                      | Example                                                                                 |
| ------------------------------------- | ------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | --------------------------------- | --------------------------------------------------------------------------------------- |

> | parLawId    | string   | Id of the Log Analytics Workspace                                                             | Mandatory input, default: `false` | `false`                                                                                 |

Reviewer: ```suggestion
| parLogAnalyticsWorkspaceResourceId | string   | Resource ID of the Log Analytics Workspace                                                             | Mandatory input | `/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/alz-logging/providers/Microsoft.OperationalInsights/workspaces/alz-log-analytics`                                                                                 |
```

## Outputs

> The module will not generate any outputs.

Reviewer: ```suggestion
*The module will not generate any outputs.*
```

## Deployment

This module requires being called with the  scope: managementGroup(mgId) function. This is done from within the orchestration module mgDiagSettingsAll.bicep module for all the management groups previously created and based on the input parameters for that parent module.

> The Diagnostic Settings resource will be named toLa but can be changed in the module if desired.

Reviewer: Can you update this section to include the PowerShell and Azure CLI examples as per all the other modules please?

e.g. https://github.com/Azure/ALZ-Bicep/tree/main/infra-as-code/bicep/modules/subscriptionPlacement#deployment

Author: @jtracey93 , I had started working on that but because this module is only used when called from the other module (because the way in which the managementgroup(mg.id) function works I believe it cant be called from PowerShell or Azure CLI. That is what I meant on my note but lets discuss, I might be wrong.

Reviewer: We discussed this 👍

## Validation

> To validate if Diagnostic Settings was correctly enabled for any specific management group, a REST API GET call can be used. Documentation and easy way to try this can be found in this link [(Management Group Diagnostic Settings - Get)](https://learn.microsoft.com/rest/api/monitor/management-group-diagnostic-settings/get?tabs=HTTP&tryIt=true&source=docs#code-try-0).

Reviewer: Add a note here as to why this is required. Because the portal experience does not show these 👍
